### PR TITLE
playercntl: Command /resetrep - to reset all reputations (12.10.20)

### PR DESCRIPTION
**Pull request by dd84ai**
Descritiption: when /resetrep is used. It will nullify all reputations to zero like in fresh restart
What is it for? to restore broken reputations after changing IDs.

Restrictions for usage: 
1) It can be used only in base + it kicks (for logging to rephack your reps according to your ID)
2) It takes some credits (default price 10'000'000)
3) It can be used only once in two weeks, cooldown is resistant to renameme and movechar operations.

### CODE REVIEW

**Comment by PhoenixxClaw** [comment]
First of all, sorry for jumping in here. :D

I'd recommend you to recode this a bit @dd84ai ... If I understand your code correctly you're basically setting the reputation of all currently existing factions to 0.0 with 'HkSetRep(wscCharname, stows(factionName), 0.0f);'. Problem with that is that you'd set this for factions who are supposed to be hostile as well. (means a Nomad IDed Character would suddenly be able to be neutral to all factions).  

I recommend adding a check for the currently equipped ID of the ship, get the default rep for it and set the rep to the defaults.

**Comment by RnScully** [comment]
I think a way to reset rep would be very helpful to many players. Changing IDs can get you into very weird places.

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 1]
```diff
@@ -252,6 +264,178 @@ namespace MiscCmds
...
+					while (ini.read_value())
+					{
+						if (ini.is_value("shipname"))
+						{
+							ini_get_wstring(ini, nickname);
+						}
+						else if (ini.is_value("timelimit"))
+						{
+							ids_name = ini.get_value_int(0);
+							ShipnamesWhichAlreadyUsedResetRep[nickname] = ids_name;
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:324]: this should be done after the `while` block, it will otherwise break if someone tries to specify  `timelimit` before `shipname`

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 2]
```diff
@@ -252,6 +264,178 @@ namespace MiscCmds
...
+
+		if (ShipnamesWhichAlreadyUsedResetRep.find(wscCharFileName) != ShipnamesWhichAlreadyUsedResetRep.end())
+		{
+			uint LastTime = ShipnamesWhichAlreadyUsedResetRep[wscCharFileName];
+
+			if (((int)time(0) - LastTime) < set_iResetrepTimeLimit)
+			{
+				int SecondsLeft = (set_iResetrepTimeLimit - ((int)time(0) - int(LastTime)));
+				int DaysLeft = SecondsLeft / (3600 * 24);
+			PrintUserCmdText(iClientID, L"ERR Resetrep time limit. %d days (%d in seconds) time remains to wait.", DaysLeft, SecondsLeft);
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:403]: broken indentation. also this could be worded better

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 3]
```diff
@@ -252,6 +264,178 @@ namespace MiscCmds
...
+			}
+		}
+
+		int count = 0;
+		for (map<string, uint>::iterator iter = factions.begin(); iter != factions.end(); iter++)
+		{
+			float fRep = 0.0f;
+			string factionName = iter->first;
+			HK_ERROR error;
+			if ((error = HkGetRep(wscCharname, stows(factionName), fRep)) != HKE_OK)
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:414]: what purpose does all this serve exactly?
do we want to check fRep != 0.0f before count++?

File: Plugins/Public/playercntl_plugin/Rename.cpp [hunk 4]
```diff
@@ -455,6 +458,15 @@ namespace Rename
 
 				// The rename worked. Log it and save the rename time.
 				AddLog("NOTICE: User rename %s to %s (%s)", wstos(o.wscCharname).c_str(), wstos(o.wscNewCharname).c_str(), wstos(HkGetAccountID(acc)).c_str());
+
+				wstring wscCharFileName2;
+				HkGetCharFileName(o.wscNewCharname, wscCharFileName2);
+				//Anti renameme, for /resetrep cooldown
+				if (MiscCmds::ShipnamesWhichAlreadyUsedResetRep.find(wscCharFileName) != MiscCmds::ShipnamesWhichAlreadyUsedResetRep.end())
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Rename.cpp:465]: So we're actually using char file names in this map instead of char names?

**Review by dsyalex — CHANGES_REQUESTED** [review]
Storing a central DB of state like this in flhook_plugins feels wrong. Don't we normally store per-char file info in files in the relevant account directory and move those around?

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 3 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:414]: Just a check for possible errors. To verify that every used factionName belongs to some registered faction.
If it would not pass, usage of HkSetRep can be not recommended... for more stable command work probably will be better having "continue;"  instead of "return true" in error catching if construction. Changed to "continue" to skip current for loop step.

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 1 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:324]: fixed

File: Plugins/Public/playercntl_plugin/Rename.cpp (hunk 4 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/Rename.cpp:465]: yup, more reliable. char file names are having always names like abdfbd_dfdsf
only English alphabet with one "_"
I feel it is more secure to use than all the random things a regular player name can be.

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 2 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:403]: fixed indentation. no idea how to reword for now.

**Comment by dd84ai** [comment]
> Storing a central DB of state like this in flhook_plugins feels wrong. Don't we normally store per-char file info in files in the relevant account directory and move those around?

yeah, you right, it would be much better. I just... presumed that central DB is good to avoid /movechar cooldown removal. But then I realized that I just can update my cooldown in during movechar operation. Rewriting.

**Comment by dd84ai** [comment]
in progres.... not done yet

**Comment by dd84ai** [comment]
Rewritten to decentralize database. Data is stored in account folders now.

**Comment by dd84ai** [comment]
https://discoverygc.com/forums/showthread.php?tid=183637
gathered opinions

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 2 above)
Comment (dsyalex) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp]: `PrintUserCmdText(iClientID, L"ERR Attempt to resetrep too frequently. Please wait %d days (%d in seconds).", DaysLeft, SecondsLeft);`

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 3 above)
Comment (dsyalex) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:420]: Given that this is data from initialworld isn't this effectively dead code that can never be hit?

File: Plugins/Public/playercntl_plugin/Rename.cpp (hunk 4 above)
Comment (dsyalex) [reply Plugins/Public/playercntl_plugin/Rename.cpp]: okay so we should name it that way

**Comment by dsyalex** [comment]
Thanks, replies above

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 2 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp]: fixed

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 3 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:420]: Yes, but.... moderators can mess up this data with being still working
This check makes sure that "SetRep" is not applied to entity which can't be changed
saving the game from possible crash to just skipped loop cycle.

This check is used very rarely anyway, only when player uses the command
So it should be as crash-proof as possible in my opinion.

File: Plugins/Public/playercntl_plugin/Rename.cpp (hunk 4 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/Rename.cpp]: it was already named as CharFileName. Not sure what you mean here

**Comment by dd84ai** [comment]
Everything is replied

File: Plugins/Public/playercntl_plugin/Rename.cpp (hunk 4 above)
Comment (dsyalex) [reply Plugins/Public/playercntl_plugin/Rename.cpp]: Right now this is named 'shipnames' (i.e., character names), but actually it's char file names. Not the same thing.

**Comment by dsyalex** [comment]
another comment above

File: Plugins/Public/playercntl_plugin/Rename.cpp (hunk 4 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/Rename.cpp]: Uh. I guess I was blind to it. Fixed.
https://github.com/DiscoveryGC/FLHook/pull/138/commits/fcc87ffc1733218a9972786c512407c5ca6b9ed9

**Comment by dd84ai** [comment]
Resolved

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 5]
```diff
@@ -64,6 +64,13 @@ namespace MiscCmds
 	/// Cost to drop reputation changes.
 	int set_iRepdropCost = 0;
 
+	/// Cost to drop all reputation changes.
+	int set_iResetdropCost = 0;
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:68]: set_iResetrepCost

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 6]
```diff
@@ -64,6 +64,13 @@ namespace MiscCmds
...
+	// Resetrep is not allowed until server restart
+	map<string, uint> factions;
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:70]: `factions` is a really misleading name for this

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 7]
```diff
@@ -252,6 +263,185 @@ namespace MiscCmds
...
+			}
+			ini.close();
+			ConPrint(L"Resetrep: Loaded %u factions\n", factions.size());
+		}
+	}
+	
+	map<wstring, uint> MiscCmds::Resetrep_Time_limits_load_map(string filename)
+	{
+		INI_Reader ini;
+		string factionpropfile = filename;
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:302]: this doesn't seem to be necessary

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 8]
```diff
@@ -252,6 +263,185 @@ namespace MiscCmds
...
+				}
+			}
+			ini.close();
+		}
+		return tempmap;
+	}
+
+	void MiscCmds::Resetrep_Time_limits_add(string filename, map<wstring, uint> tempmap)
+	{
+		string factionpropfile = filename;
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:334]: this doesn't seem to be necessary

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 9]
```diff
@@ -252,6 +263,185 @@ namespace MiscCmds
...
+				{
+					uint timelimit = 0;
+					wstring nickname = L"0";
+					while (ini.read_value())
+					{
+						if (ini.is_value("shipname"))
+						{
+							ini_get_wstring(ini, nickname);
+						}
+						else if (ini.is_value("timelimit"))
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:318]: shouldn't this be named more like `lastresettime`?

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 10]
```diff
@@ -252,6 +263,185 @@ namespace MiscCmds
...
+	{
+		string factionpropfile = filename;
+
+		FILE *file = fopen(factionpropfile.c_str(), "w");
+		if (file)
+		{
+			for (map<wstring, uint>::iterator i = tempmap.begin();
+				i != tempmap.end(); ++i)
+			{
+				if (((int)time(0) - i->second) < set_iResetrepTimeLimit)
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:342]: wait why is this here??

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 11]
```diff
@@ -252,6 +263,185 @@ namespace MiscCmds
...
+			if ((error = HkGetRep(wscCharname, stows(factionName), fRep)) != HKE_OK)
+			{
+				PrintUserCmdText(iClientID, L"ERR (5) %s", HkErrGetText(error).c_str());
+				continue;
+			}
+			count++;
+			HkSetRep(wscCharname, stows(factionName), 0.0f);
+		}
+
+		PrintUserCmdText(iClientID, L"OK Reset Rep is performed. %d factions were cleared.", count);
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:429]:
```suggestion
		PrintUserCmdText(iClientID, L"OK Rep reset has beens performed. %d factions were cleared.", count);
```

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 12]
```diff
@@ -252,6 +263,185 @@ namespace MiscCmds
...
+
+		// Remove cash if we're charging for it.
+		if (set_iResetdropCost > 0)
+		{
+			HkAddCash(wscCharname, 0 - set_iResetdropCost);
+		}
+
+		ShipnamesWhichAlreadyUsedResetRep[wscCharFileName] = (int)time(0);
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:437]: maybe name it CharfilesLastResetRepUsage?

File: Plugins/Public/playercntl_plugin/Rename.cpp [hunk 13]
```diff
@@ -455,6 +463,26 @@ namespace Rename
...
+				HkGetCharFileName(o.wscNewCharname, wscCharFileName2);
+				wstring wscDir;	
+				HK_ERROR err;
+				if ((err = HkGetAccountDirName(o.wscNewCharname, wscDir)) != HKE_OK)
+				{
+					ConPrint(L"ERR 461 " + HkErrGetText(err));
+				}
+				else
+				{
+					string scRenameFile = scAcctPath + wstos(wscDir) + "\\" + "resetrep.ini";
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Rename.cpp:478]:
```suggestion
					string scRenameFile = scAcctPath + wstos(wscDir) + "\\resetrep.ini";
```

File: Plugins/Public/playercntl_plugin/Rename.cpp [hunk 14]
```diff
@@ -873,6 +912,8 @@ namespace Rename
 		o.scSourceFile = scAcctPath + wstos(wscSourceDir) + "\\" + wstos(wscSourceFile) + ".fl";
 		o.scDestFile = scAcctPath + wstos(wscDir) + "\\" + wstos(wscSourceFile) + ".fl";
 		o.scDestFileTemp = scAcctPath + wstos(wscDir) + "\\" + wstos(wscSourceFile) + ".fl.moving";
+		o.scSourceDir = scAcctPath + wstos(wscSourceDir) + "\\" + "resetrep.ini";
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Rename.cpp:915]:
```suggestion
		o.scSourceDir = scAcctPath + wstos(wscSourceDir) + "\\resetrep.ini";
```

File: Plugins/Public/playercntl_plugin/Rename.cpp [hunk 15]
```diff
@@ -873,6 +912,8 @@ namespace Rename
...
+		o.scDestDir = scAcctPath + wstos(wscDir) + "\\" + "resetrep.ini";
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Rename.cpp:916]:
```suggestion
		o.scDestDir = scAcctPath + wstos(wscDir) + "\\resetrep.ini";
```

**Review by dsyalex — CHANGES_REQUESTED** [review]

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 10 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:342]: if only I could remember...

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 7 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:302]: it looks like a function that load 'cooldown restrictions' for certain player folder/account

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 8 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:334]: it looks like a function that rewrites/saves player time limits for all present charfilenames in some player folder

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 10 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:342]: from what I was able to find... in the code which I certainly do not remember anymore

those two extra functions were made as 'wrappers' to deal with time limits being stored in files.

it looks to be very useful when 'data' about time limits needs to be 'loaded' again in rename and move char functions
image

Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:342]: more precide namings for those functions would be
Resetrep_load_time_limit_for_player account(folder)
Resetrep_save_time_limits_for_player_account(folder)

Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:342]: After re-reading this code, I still think those functions are still needed. Because I don't want to write fully code to 'read' data from files for all three times it is needed

Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:342]: I could make it more looking like a one class wrapping those dealings though.

something like
class time_limits_for_player_account
{
map_of_time_limits
construct()
{ load from file }
save()
{ save to file }
get_time_limits()
{ return map_of_time_limits }
}

Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:342]: Resetrep_save_time_limits_for_player_account is used three times, good enough to be a function and not copy pasted code

Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:342]: commited and squashed renames to functions

**Comment by dd84ai** [comment]
answered

**Comment by dd84ai** [comment]
Doing tests:
1) I had hostile reps, did /resetrep, was kicked, reps aren't hostile anymore
2) Attempt to do second /resetrep, denied and informed to wait 13 days (1209538 seconds)
2) Attempt to do third /resetrep, denied and informed to wait 13 days (1209501 seconds)
3) Edited by server side files to change last time of resetrep to 2'000'000 seconds into the past, attempted resetrep second time, succeded
4) ....test fail.

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 7 above)
Comment (dsyalex) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:302]: right but why not just either rename `filename` to `factionpropfile`, or drop all references to `factionpropfile` and use `filename`?

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 8 above)
Comment (dsyalex) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:334]: see above

**Review by dsyalex — CHANGES_REQUESTED** [review]

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 7 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:302]: indeed.

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 8 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:334]: understood the issue

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 16]
```diff
@@ -64,6 +64,13 @@ namespace MiscCmds
 	/// Cost to drop reputation changes.
 	int set_iRepdropCost = 0;
 
+	/// Cost to drop all reputation changes.
+	int set_iResetrepCost = 0;
+	// Resetrep is not allowed until server restart
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:69]: this doc comment is not related to the variable below it

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 17]
```diff
@@ -252,6 +263,183 @@ namespace MiscCmds
...
+		INI_Reader ini;
+		map<wstring, uint> tempmap;
+		if (ini.open(filename.c_str(), false))
+		{
+			while (ini.read_header())
+			{
+				if (ini.is_header("Playership"))
+				{
+					uint lastresettime = 0;
+					wstring nickname = L"0";
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:310]: weird default ship name? nickname is also a weird name.

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 18]
```diff
@@ -252,6 +263,183 @@ namespace MiscCmds
...
+		if (file)
+		{
+			for (map<wstring, uint>::iterator i = tempmap.begin();
+				i != tempmap.end(); ++i)
+			{
+				if (((int)time(0) - i->second) < set_iResetrepTimeLimit)
+				{
+					wstring temp = i->first;
+					fprintf_s(file, "[Playership]\n");
+					ini_write_wstring(file, "shipname", temp);
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:344]: directly use i->first ?

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 10 above)
Comment (dsyalex) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:340]: I still don't understand why we're checking the current time inside a function intended to write a map to disk.

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 19]
```diff
@@ -252,6 +263,183 @@ namespace MiscCmds
...
+		{
+			PrintUserCmdText(iClientID, L"ERR Insufficient credits");
+			return true;
+		}
+
+
+		wstring wscCharFileName;
+		if ((err = HkGetCharFileName(wscCharname, wscCharFileName)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR (2)" + HkErrGetText(err));
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:386]: why do we concatenate here but do string formatting and c_str above? also missing space after `(2)`?

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 20]
```diff
@@ -252,6 +263,183 @@ namespace MiscCmds
...
+			return true;
+		}
+
+		// Read the last time a rename was done on this character
+		wstring wscDir;
+		if ((err = HkGetAccountDirName(wscCharname, wscDir)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR (3)" + HkErrGetText(err));
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:394]: see above

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 21]
```diff
@@ -252,6 +263,183 @@ namespace MiscCmds
...
+			if ((error = HkGetRep(wscCharname, stows(factionName), fRep)) != HKE_OK)
+			{
+				PrintUserCmdText(iClientID, L"ERR (5) %s", HkErrGetText(error).c_str());
+				continue;
+			}
+			count++;
+			HkSetRep(wscCharname, stows(factionName), 0.0f);
+		}
+
+		PrintUserCmdText(iClientID, L"OK Rep reset has beens performed. %d factions were cleared.", count);
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:427]:
```suggestion
		PrintUserCmdText(iClientID, L"OK Rep reset has been performed. %d factions were cleared.", count);
```

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 22]
```diff
@@ -252,6 +263,183 @@ namespace MiscCmds
...
+
+		int count = 0;
+		for (map<string, uint>::iterator iter = factions.begin(); iter != factions.end(); iter++)
+		{
+			float fRep = 0.0f;
+			string factionName = iter->first;
+			HK_ERROR error;
+			if ((error = HkGetRep(wscCharname, stows(factionName), fRep)) != HKE_OK)
+			{
+				PrintUserCmdText(iClientID, L"ERR (5) %s", HkErrGetText(error).c_str());
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:420]: we missed `(4)`?

File: Plugins/Public/playercntl_plugin/Rename.cpp [hunk 23]
```diff
@@ -455,6 +463,26 @@ namespace Rename
...
 				AddLog("NOTICE: User rename %s to %s (%s)", wstos(o.wscCharname).c_str(), wstos(o.wscNewCharname).c_str(), wstos(HkGetAccountID(acc)).c_str());
+
+				//resetrep cooldown
+				wstring wscCharFileName2;
+				HkGetCharFileName(o.wscNewCharname, wscCharFileName2);
+				wstring wscDir;	
+				HK_ERROR err;
+				if ((err = HkGetAccountDirName(o.wscNewCharname, wscDir)) != HKE_OK)
+				{
+					ConPrint(L"ERR 461 " + HkErrGetText(err));
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Rename.cpp:474]: okay we could do something better than 461 here if it's printing errors to console - like the function name and that it's an HkGetAccountDirName check for resetrep. But also what about logging to disk?

**Review by dsyalex — CHANGES_REQUESTED** [review]

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 17 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:310]: ship_filename willl be better I guess

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 18 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:344]: Sure

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 19 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:386]: >I still don't understand why we're checking the current time inside a function intended to write a map to disk.
This was intended... to write to disk only time limits which have not expired yet

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 20 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:394]: renamed

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 19 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:386]: renamed error

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 21 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:427]: Ok

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 22 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:420]: Renamed

File: Plugins/Public/playercntl_plugin/Rename.cpp (hunk 23 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/Rename.cpp:474]: Uh, they aren't ever should happen at all. So no need to log

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 16 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp]: Fixed!

**Comment by dd84ai** [comment]
`I still don't understand why we're checking the current time inside a function intended to write a map to disk.`
This was intended... to write to disk only time limits which have not expired yet

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 10 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:343]: This was intended... to write to disk only time limits which have not expired yet
nvm, answering in github

**Comment by dd84ai** [comment]
`I still don't understand why we're checking the current time inside a function intended to write a map to disk.`

This was intended... to write to disk only time limits which have not expired yet

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp [hunk 24]
```diff
@@ -252,6 +265,184 @@ namespace MiscCmds
...
+			float fRep = 0.0f;
+			string factionName = iter->first;
+			HK_ERROR error;
+			if ((error = HkGetRep(wscCharname, stows(factionName), fRep)) != HKE_OK)
+			{
+				PrintUserCmdText(iClientID, L"ERR Resetrep, check for faction failed");
+				continue;
+			}
+			count++;
+			HkSetRep(wscCharname, stows(factionName), 0.0f);
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/MiscCmds.cpp:427]: just sets all factions to neutral? shall we leave a comment saying setting all factions to neutral is safe because the player must be docked and rep hacks will take effect on undock?

File: Plugins/Public/playercntl_plugin/Rename.cpp [hunk 25]
```diff
@@ -455,6 +463,26 @@ namespace Rename
...
 				AddLog("NOTICE: User rename %s to %s (%s)", wstos(o.wscCharname).c_str(), wstos(o.wscNewCharname).c_str(), wstos(HkGetAccountID(acc)).c_str());
+
+				//resetrep cooldown
+				wstring wscCharFileName2;
+				HkGetCharFileName(o.wscNewCharname, wscCharFileName2);
+				wstring wscDir;	
+				HK_ERROR err;
+				if ((err = HkGetAccountDirName(o.wscNewCharname, wscDir)) != HKE_OK)
+				{
+					ConPrint(L"ERR Resetrep failure in rename to get directory");
```
Comment (dsyalex) [inline Plugins/Public/playercntl_plugin/Rename.cpp:474]: missing newline?

**Review by dsyalex — CHANGES_REQUESTED** [review]
almost there

File: Plugins/Public/playercntl_plugin/Rename.cpp (hunk 25 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/Rename.cpp:474]: Right

File: Plugins/Public/playercntl_plugin/MiscCmds.cpp (hunk 24 above)
Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:427]: All right.

Comment (dd84ai) [reply Plugins/Public/playercntl_plugin/MiscCmds.cpp:430]: Added
//We reset reps to zero, but it is safe because the ship is docked and kicked
//So rephacks take effect right after logging back

**Review by dsyalex — APPROVED** [review]
